ALGORITHMS.md

```diff
@@ -164,48 +188,58 @@ If [OQS_KEM_ENCODERS](CONFIGURE.md#OQS_KEM_ENCODERS) is enabled the following li
-| p521_frodo1344aes | 1.3.9999.99.57 | OQS_OID_P521_FRODO1344AES
-| frodo1344shake | 1.3.9999.99.60 | OQS_OID_FRODO1344SHAKE
-| p521_frodo1344shake | 1.3.9999.99.59 | OQS_OID_P521_FRODO1344SHAKE
+| frodo640aes | 1.3.9999.99.58 | OQS_OID_FRODO640AES
+| p256_frodo640aes | 1.3.9999.99.57 | OQS_OID_P256_FRODO640AES
+| x25519_frodo640aes | 1.3.9999.99.43 | OQS_OID_X25519_FRODO640AES
+| frodo640shake | 1.3.9999.99.60 | OQS_OID_FRODO640SHAKE
+| p256_frodo640shake | 1.3.9999.99.59 | OQS_OID_P256_FRODO640SHAKE
+| x25519_frodo640shake | 1.3.9999.99.44 | OQS_OID_X25519_FRODO640SHAKE
+| frodo976aes | 1.3.9999.99.62 | OQS_OID_FRODO976AES
+| p384_frodo976aes | 1.3.9999.99.61 | OQS_OID_P384_FRODO976AES
+| x448_frodo976aes | 1.3.9999.99.45 | OQS_OID_X448_FRODO976AES
+| frodo976shake | 1.3.9999.99.64 | OQS_OID_FRODO976SHAKE
+| p384_frodo976shake | 1.3.9999.99.63 | OQS_OID_P384_FRODO976SHAKE
+| x448_frodo976shake | 1.3.9999.99.46 | OQS_OID_X448_FRODO976SHAKE
+| frodo1344aes | 1.3.9999.99.66 | OQS_OID_FRODO1344AES
+| p521_frodo1344aes | 1.3.9999.99.65 | OQS_OID_P521_FRODO1344AES
+| frodo1344shake | 1.3.9999.99.68 | OQS_OID_FRODO1344SHAKE
+| p521_frodo1344shake | 1.3.9999.99.67 | OQS_OID_P521_FRODO1344SHAKE
 | kyber512 | 1.3.6.1.4.1.22554.5.6.1 | OQS_OID_KYBER512
 | p256_kyber512 | 1.3.6.1.4.1.22554.5.7.1 | OQS_OID_P256_KYBER512
 | x25519_kyber512 | 1.3.6.1.4.1.22554.5.8.1 | OQS_OID_X25519_KYBER512
 | kyber768 | 1.3.6.1.4.1.22554.5.6.2 | OQS_OID_KYBER768
-| p384_kyber768 | 1.3.9999.99.61 | OQS_OID_P384_KYBER768
-| x448_kyber768 | 1.3.9999.99.42 | OQS_OID_X448_KYBER768
-| x25519_kyber768 | 1.3.9999.99.43 | OQS_OID_X25519_KYBER768
-| p256_kyber768 | 1.3.9999.99.44 | OQS_OID_P256_KYBER768
+| p384_kyber768 | 1.3.9999.99.69 | OQS_OID_P384_KYBER768
+| x448_kyber768 | 1.3.9999.99.47 | OQS_OID_X448_KYBER768
+| x25519_kyber768 | 1.3.9999.99.48 | OQS_OID_X25519_KYBER768
+| p256_kyber768 | 1.3.9999.99.49 | OQS_OID_P256_KYBER768
 | kyber1024 | 1.3.6.1.4.1.22554.5.6.3 | OQS_OID_KYBER1024
-| p521_kyber1024 | 1.3.9999.99.62 | OQS_OID_P521_KYBER1024
-| bikel1 | 1.3.9999.99.64 | OQS_OID_BIKEL1
-| p256_bikel1 | 1.3.9999.99.63 | OQS_OID_P256_BIKEL1
-| x25519_bikel1 | 1.3.9999.99.45 | OQS_OID_X25519_BIKEL1
-| bikel3 | 1.3.9999.99.66 | OQS_OID_BIKEL3
-| p384_bikel3 | 1.3.9999.99.65 | OQS_OID_P384_BIKEL3
-| x448_bikel3 | 1.3.9999.99.46 | OQS_OID_X448_BIKEL3
-| bikel5 | 1.3.9999.99.68 | OQS_OID_BIKEL5
-| p521_bikel5 | 1.3.9999.99.67 | OQS_OID_P521_BIKEL5
-| hqc128 | 1.3.9999.99.70 | OQS_OID_HQC128
-| p256_hqc128 | 1.3.9999.99.69 | OQS_OID_P256_HQC128
-| x25519_hqc128 | 1.3.9999.99.47 | OQS_OID_X25519_HQC128
-| hqc192 | 1.3.9999.99.72 | OQS_OID_HQC192
-| p384_hqc192 | 1.3.9999.99.71 | OQS_OID_P384_HQC192
-| x448_hqc192 | 1.3.9999.99.48 | OQS_OID_X448_HQC192
-| hqc256 | 1.3.9999.99.74 | OQS_OID_HQC256
-| p521_hqc256 | 1.3.9999.99.73 | OQS_OID_P521_HQC256
+| p521_kyber1024 | 1.3.9999.99.70 | OQS_OID_P521_KYBER1024
+| mlkem512 | 1.3.6.1.4.1.22554.5.6.1 | OQS_OID_MLKEM512
```

Comment:
ML-KEM (e.g., mlkem512) has the same OID as Kyber (e.g., kyber512), is this intentional?

Reply:
Kind-of: In the absence of a spec (or can you point to one, @bhess ?) I took [the OIDs chosen by the IETF hackathon](https://github.com/IETF-Hackathon/pqc-certificates/blob/master/docs/oid_mapping.md) (assuming that's the latest ... is it, @praveksharma ?) And if they chose the same OID as for Kyber, well, that points to bad OID management (in this case apparently by the Legions of Bouncy Castle owning that range) and/or the need for someone to take the lead and draft a spec (@dstebila -- what about an "independent" PQ alliance taking that lead?)

Reply:
> Kind-of: In the absence of a spec (or can you point to one, @bhess ?) I took [the OIDs chosen by the IETF hackathon](https://github.com/IETF-Hackathon/pqc-certificates/blob/master/docs/oid_mapping.md) (assuming that's the latest ... is it, @praveksharma ?) And if they chose the same OID as for Kyber, well, that points to bad OID management (in this case apparently by the Legions of Bouncy Castle owning that range) and/or the need for someone to take the lead and draft a spec (@dstebila -- what about an "independent" PQ alliance taking that lead?)

I don't know who is meant to assign the permanent OID for an algorithm -- is it the inventor? the standards body? the first implementer? an alliance?  @bhess, if memory serves, IBM assigned some OIDs for earlier rounds of Kyber and Dilithium, are you planning to assign OIDs for ML-*-ipd or ML-*?

Reply:
NIST usually registers OIDs for approved/standardized algorithms in [CSOR](https://csrc.nist.gov/projects/computer-security-objects-register/algorithm-registration). We have in the past reserved OIDs for algorithms used but not already assigned (earlier round versions and ML-DSA-ipd).

Reply:
> We have in the past reserved OIDs for algorithms used

Please document this in generate.yml as per #351; more people taking responsibility for proper (O)ID allocation very welcome: We could split this task among all of us, e.g., along algorithms: Volunteers welcome!